eclass/flag-o-matic.eclass

```diff
@@ -372,19 +380,30 @@ replace-cpu-flags() {
 # @USAGE: <variable> <flag>
 # @INTERNAL
 # @DESCRIPTION:
-# Returns shell true if <flag> is in a given <variable>, else returns shell false.
+# Returns shell true if <flag> is in a given <variable> and not surpassed by e.g. -fno-<flag>,
+# else returns shell false.
 _is_flagq() {
-	local x var="$1[*]"
+	local x var="$1[*]" equal_str equal_pos flag_no active=1
+	equal_str=${2%%=*}
+	equal_pos=${#equal_str}
+	# This results in -fno-flag for -fflag=*
+	flag_no="${2:0:2}no-${2:2:(equal_pos - 2)}"
+
 	for x in ${!var} ; do
-		[[ ${x} == $2 ]] && return 0
+		case ${x} in
+			$2) active=0 ;;
+			$flag_no) active=1 ;;
```

Comment:
I'd personally rename active to `ret` or something, active=1 meaning that it's not active feels kind of confusing. Not that ret says much about what it means, but well.